src/celeritas/phys/detail/PreStepActionImpl.hh

```diff
@@ -62,10 +62,20 @@ inline CELER_FUNCTION void pre_step_track(celeritas::CoreTrackView const& track)
         step.element({});
     }
 
-    // Sample mean free path
     auto phys = track.make_physics_view();
+    if (phys.num_particle_processes() == 0 && phys.scalars().transport)
+    {
+        // Replicate G4Transportation
+        // Set step limit to infinity and exit
+        sim.reset_step_limit();
```

Comment:
Instead of this, why not set to `{geo boundary, numeric_limits::max}`? I think then you wouldn't have to make any changes to the along-step kernel.